### ISSUE TITLE
Bug in `isinstance(v, Sized)`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1741]
```python
from typing import NoReturn, Sized

def f(a: list[int]):
    assert isinstance(a, Sized)
    _: NoReturn = a  # Should Error, but No Error (pyright infers `a: Never`)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version.

**Comment by erictraut** [comment]
This is included in Pyright 1.1.130, which I just published. It will also be part of the next release of Pylance.
